File: Solutions/C#/104-maximum-depth-of-binary-tree.cs
/*
Problem Title: 104 - Maximum Depth of Binary Tree
Lang: C#
Runtime: 96 ms
Memory: 25.4 MB
Problem Url: https://leetcode.com/problems/maximum-depth-of-binary-tree
Submission Url: https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/303445201
*/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int MaxDepth(TreeNode root) {
        if(root == null){
            return 0;
        }
        return Math.Max(MaxDepth(root.left), MaxDepth(root.right)) + 1;
    }
}

File: Solutions/C++/Accepted/104-maximum-depth-of-binary-tree.cpp
/*
Problem Title: 104 - Maximum Depth of Binary Tree
Lang: C++
Runtime: 10 ms
Memory: 18.9 MB
Problem Url: https://leetcode.com/problems/maximum-depth-of-binary-tree
Submission Url: https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/898823247
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == NULL){
            return 0;
        }

        return max(maxDepth(root->left) ,maxDepth(root->right) )+1; 
    }
};